input is declared with attribute nonnull -- causing a
warning for the superfluous NULL check with clang
in code we want to build with -Werror
--- libxcrypt-3.1.1/src/xcrypt.c.omv~	2018-01-24 11:46:23.756126148 +0100
+++ libxcrypt-3.1.1/src/xcrypt.c	2018-01-24 11:46:52.007120384 +0100
@@ -170,13 +170,6 @@ __xcrypt_gensalt_r (__const char *prefix
   char *(*use) (unsigned long count,
 		__const char *input, int size, char *output, int output_size);
 
-  /* This may be supported on some platforms in the future */
-  if (!input)
-    {
-      __set_errno (EINVAL);
-      return NULL;
-    }
-
   if (prefix[0] == '$')
     {
       char *hash_id = strdup (&prefix[1]);
--- libxcrypt-3.1.1/configure.ac.omv~	2018-01-24 11:50:24.683143626 +0100
+++ libxcrypt-3.1.1/configure.ac	2018-01-24 11:52:43.871208780 +0100
@@ -93,6 +93,26 @@ if eval "test x$GCC = xyes"; then
         EXTRA_CFLAGS="$EXTRA_CFLAGS -Wno-cast-align"
     fi
 fi
+
+AC_ARG_ENABLE(
+    [Wno-null-pointer-arithmetic],
+    AS_HELP_STRING(
+        [--enable-Wno-null-pointer-arithmetic],
+        [do not warn about cast increases required alignment of target type]
+    ),
+    [case "${enableval}" in
+        yes) no_cast_align=true ;;
+         no) no_cast_align=false ;;
+          *) AC_MSG_ERROR([bad value ${enableval} for --enable-Wno-null-pointer-arithmetic]) ;;
+     esac],
+     [no_cast_align=false]
+)
+
+if eval "test x$GCC = xyes"; then
+    if eval "test x$no_cast_align = xtrue"; then
+        EXTRA_CFLAGS="$EXTRA_CFLAGS -Wno-null-pointer-arithmetic"
+    fi
+fi
 AC_SUBST(EXTRA_CFLAGS)
 
 AC_CONFIG_FILES([Makefile src/Makefile plugins/Makefile plugins/blowfish/Makefile
